### PR TITLE
Bookstore Components,Reducers,Actions

### PR DESCRIPTION
**Pull request by Alexoid1**
Add reducer,actions,and component
Add prop-types to components props
Add redux and react connect

### CODE REVIEW

File: src/action-types/index.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const REMOVE_BOOK = 'REMOVE_BOOK';
+export const CREATE_BOOK = 'CREATE_BOOK';
```
Comment (bellom) [inline src/action-types/index.js:2]: I think you already created this in `src/actions/action-types.js.`
kindly remove.

File: src/components/App.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import './App.css';
+import BooksForm from './BooksForm';
+import BooksList from './BooksList';
 
 function App() {
   return (
     <div className="App">
-      <h1>Hello!</h1>
+      <BooksForm />
+      <BooksList />
```
Comment (bellom) [inline src/components/App.js]: ### [Optional]
You could use react fragment here, just the way you used it in `src/components/BooksForm.js`

File: src/components/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
```
Comment (bellom) [inline src/components/BooksForm.js]: Let's do some rearrangement.
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory, (presentational components) you have components that are not connected to the Redux store, and on your containers directory you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksList.js]: Let's do some rearrangement.
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory, (presentational components) you have components that are not connected to the Redux store, and on your containers directory you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/actions/action-types.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (bellom) [inline src/actions/action-types.js]: According to the project instructions on pathwright. you're supposed to use `src/actions/index.js`

Kindly correct the path, this looks nice but to be more professional about this, it good to abide by the React project structure.
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Alexoid1  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

**Let's work on GitHub Action;**
Microverse has moved on to Github actions for continuous deployment 
image

https://github.com/microverseinc/linters-config/tree/master/javascript

**Add linter's check:**
checkout this [linters-config#troubleshooting](https://github.com/microverseinc/linters-config#troubleshooting) for help. Example below;
image
If you are not having this check on your `Github` page, kindly visit :point_right:[ linter set up](https://github.com/microverseinc/linters-config/tree/master/react-redux)

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by whiz25 — APPROVED** [review]
## Approved :clap:
Well done @Alexoid1 . You made the requested changes. You moved `BooksForm.js` from`components` folder to the `containers` folder as requested.
Merge and continue to the next project.


### Martin Afani

**Pull request merged**